libs/sdk-ui-pivot/src/impl/data/rowFactory.ts

```diff
@@ -0,0 +1,229 @@
+// (C) 2007-2021 GoodData Corporation
+import { IntlShape } from "react-intl";
+
+import { DataViewFacade, IMappingHeader } from "@gooddata/sdk-ui";
+import { ROW_SUBTOTAL, ROW_TOTAL } from "../base/constants";
+import {
+    DataValue,
+    IResultHeader,
+    isResultAttributeHeader,
+    isResultTotalHeader,
+} from "@gooddata/sdk-backend-spi";
+import invariant from "ts-invariant";
+import { isDataColLeaf, SliceCol } from "../structure/tableDescriptorTypes";
+import { TableDescriptor } from "../structure/tableDescriptor";
+import { IAgGridPage, IGridRow, IGridTotalsRow } from "./resultTypes";
+import { getSubtotalStyles } from "./dataSourceUtils";
+
+function getSubtotalLabelCellIndex(resultHeaderItems: IResultHeader[][], rowIndex: number): number {
+    return resultHeaderItems.findIndex((headerItem) => isResultTotalHeader(headerItem[rowIndex]));
+}
+
+function getMinimalRowData(dv: DataViewFacade): DataValue[][] {
+    const data = dv.rawData().twoDimData();
+    const rowHeaders = dv.meta().allHeaders()[0];
+    const numberOfRowHeaderItems = (rowHeaders[0] || []).length;
+
+    return data.length > 0
+        ? data
+        : // if there are no measures only attributes
+          // create array of [null] of length equal to the number of row dimension headerItems
+          (Array(numberOfRowHeaderItems).fill([null]) as DataValue[][]);
```

Comment:
`Array.prototype.fill` is not supported by IE11 and we do not polyfill this. You can use lodash.fill instead.

Reply:
interesting, this code was here for a while :)

Reply:
i'm going to check it out though

Reply:
fixed